src/components/Proposal/Create/PluginForms/CreateUnknownGenericPluginProposal.tsx

```diff
@@ -72,12 +89,23 @@ class CreateGenericPlugin extends React.Component<IProps, IStateProps> {
       () => Object.assign(this.currentFormValues, this.state),
       (formValues: IFormValues, firstTime: boolean) => {
         this.currentFormValues = formValues;
-        if (firstTime) { this.state = { tags: formValues.tags }; }
-        else { this.setState({ tags: formValues.tags }); }
+        if (firstTime) {
+          Object.assign(this.state, {
+            tags: formValues.tags, abiInputs: this.state.abiInputs, callData: this.state.callData,
+          });
+        }
+        else { this.setState({ tags: formValues.tags, abiInputs: [], callData: "" }); }
       },
       this.props.showNotification);
   }
 
+  async componentDidMount() {
+    const contractToCall = (this.props.pluginState as IGenericPluginState).pluginParams.contractToCall;
+    const ABI_DATA = await getABIByContract(contractToCall);
```

Comment:
why ABI_DATA in Capital letters ?

Reply:
No reason, will change it to `abiData` and `ABI_METHODS` to `abiMethods`.